src/java/com/puppetlabs/puppetserver/RegisteredLinkedBlockingDeque.java

```diff
@@ -40,6 +40,21 @@ synchronized public void registerLast(E e) throws InterruptedException {
     }
 
     /**
+     * This method removes an element from the list of "registered" elements,
+     * such that it will no longer be returned by calls to
+     * <tt>getRegisteredInstances</tt>.
+     *
+     * This method does not remove the element from the underlying queue; it
+     * is assumed that the caller has already done so via the methods of the
+     * parent class.
```

Comment:
Should it?  Would it be better if this method also removed the element from the underlying queue?  That would make it consistent with `register`.